v2/tools/generator/internal/config/type_transformer.go

```diff
@@ -55,37 +56,163 @@ type TypeTransformer struct {
 	matchedProperties map[astmodel.TypeName]string
 }
 
-func produceTargetType(
-	target TransformTarget,
+func (target *TransformTarget) AppliesToType(t astmodel.Type) bool {
+	if target.appliesCache == nil {
+		target.appliesCache = make(map[astmodel.Type]bool)
+	}
+
+	if result, ok := target.appliesCache[t]; ok {
+		return result
+	}
+
+	result := target.appliesToType(t)
+	target.appliesCache[t] = result
+	return result
+}
+
+func (target *TransformTarget) appliesToType(t astmodel.Type) bool {
+	if target == nil {
+		return true
+	}
+
+	inspect := t
+	if target.Optional {
+		// Need optional
+		opt, ok := astmodel.AsOptionalType(inspect)
```

Comment:
I don't think this is what `target.Optional` meant in the old code? Looking at the original `TransformTarget`, optional was used like this in `produceTargetType`:

```
if target.Optional {
	result = astmodel.NewOptionalType(result)
}
```

That's the only place it was used. Maybe you're fixing a bug here where it just was ignored when used on `ifType`? I think that's what's happening but just wanna confirm.

Reply:
That result type was used with a `Equals()` to see if we had the type that needed to be modified; this approach doesn't work with wildcards. Instead, we check to see if we need optional, and if we do, whether we have it.